internal/tx/txnprocessor.go

```diff
@@ -235,8 +249,55 @@ func (p *txnProcessor) resolveSigner(from string) (signer eth.TXSigner, err erro
 	return
 }
 
+// idempotencyCheck called by addInflightWrapper within the inflight lock, in the case the
+// extra ackType=receipt idempotency check is enabled, and possible due to co-location with
+// the REST API Gateway.
+func (p *txnProcessor) idempotencyCheck(inflight *inflightTxn, inflightForAddr *inflightTxnState) (bool, error) {
+	inflight.idempotencyCheck = true
+
+	// First check the in-memory list
+	for _, alreadyInflight := range inflightForAddr.txnsInFlight {
+		if alreadyInflight.msgID == inflight.msgID {
+			log.Warnf("Kafka redelivery of message already inflight: %s", inflight.msgID)
+			// We don't send a new reply here - special nil return
+			return false, nil
+		}
+	}
+	// Then check LevelDB - we should find the entry
+	r, err := p.receiptStore.GetReceipt(inflight.msgID)
+	if err != nil {
+		return false, err
```

Comment:
Do we need to add a richer return status to account for transient receipt store access issues? I'm wondering what the implication of indicating a false negative is for the application. Here, we are saying "we don't know if there is a receipt with this ID or not, assume this is _not_ idempotent".

Reply:
I did track through the result of this error return, and it will actually end up coming back as an error reply in Kafka, which would overwrite any "good" reply if there was one that was earlier.

I couldn't think of a better option here:
-  Infinite retry felt wrong under the lock
-  Silently ignoring felt wrong, because we can't sure sure an event went back at all in that case

Reply:
Thanks for this @vdamle - I've added a more descriptive error, but am open to other suggestions too

Reply:
Perfect, looks great now.